SR-IOV: XML refresh

adriansuhov opened the pull request:
The test case names and order was refreshed. New TC names are:
SRIOV-VERIFY-LSPCI
SRIOV-VERIFY-SINGLE-VF-CONNECTION
SRIOV-VERIFY-SINGLE-VF-CONNECTION-MAX-VCPU
SRIOV-VERIFY-MAX-VF-CONNECTION
SRIOV-VERIFY-MAX-VF-CONNECTION-MAX-VCPU
SRIOV-ADD-MAX-NICS-DURING-PROVISION
SRIOV-ADD-MAX-NICS-AFTER-PROVISION
SRIOV-DISABLE-ENABLE-AN
SRIOV-INTERRUPTS-CHANGE
SRIOV-DISABLEVF-ON-GUEST
SRIOV-RELOAD-MODULE
SRIOV-IPERF-STRESS

During this refresh, the script "verify-ifup-nics.sh" was removed and its features were integrated in NET-Max-NICs.sh. By doing this, we'll be using the same bash script for Azure and Hyper-V testing Max NICs
test cases.

# Test results - Azure
[LISAv2 Test Results Summary]
Test Run On           : 04/25/2019 08:17:59
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Test Category         : Functional
Test Area             : SRIOV
Total Test Cases      : 12 (11 Passed, 1 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:2:10

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 SRIOV                SRIOV-VERIFY-MAX-VF-CONNECTION-MAX-VCPU                                           PASS                 6.02 
    2 SRIOV                SRIOV-VERIFY-SINGLE-VF-CONNECTION                                                 PASS                 2.24 
    3 SRIOV                SRIOV-DISABLEVF-ON-GUEST                                                          PASS                 2.47 
    4 SRIOV                SRIOV-RELOAD-MODULE                                                               PASS                 2.45 
    5 SRIOV                SRIOV-IPERF-STRESS                                                                PASS                34.03 
    6 SRIOV                SRIOV-VERIFY-SINGLE-VF-CONNECTION-MAX-VCPU                                        PASS                 2.28 
    7 SRIOV                SRIOV-INTERRUPTS-CHANGE                                                           PASS                14.03 
    8 SRIOV                SRIOV-VERIFY-LSPCI                                                                PASS                 0.89 
    9 SRIOV                SRIOV-ADD-MAX-NICS-DURING-PROVISION                                               PASS                 0.83 
   10 SRIOV                SRIOV-ADD-MAX-NICS-AFTER-PROVISION                                                PASS                 9.09 
   11 SRIOV                SRIOV-DISABLE-ENABLE-AN                                                           FAIL                20.16 
	EnableSRIOV : Test Iteration - 1 : PASS 
	DisableSRIOV : Test Iteration - 1 : PASS 
	EnableSRIOV : Test Iteration - 2 : PASS 
	DisableSRIOV : Test Iteration - 2 : PASS 
	EnableSRIOV : Test Iteration - 3 : FAIL 
   12 SRIOV                SRIOV-VERIFY-MAX-VF-CONNECTION                                                    PASS                 6.04 

SRIOV-DISABLE-ENABLE-AN is known to fail on 18.04

# Test results - Hyper-V
[LISAv2 Test Results Summary]
Test Run On           : 04/24/2019 18:34:06
VHD Under Test        : ******\Ubuntu\Ubuntu_14.04.5\Ubuntu14.04.5-server.vhdx
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:3

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 SRIOV                NETWORK-ADD-MAX-SYNTHETIC-NICS-DURING-PROVISION                                   PASS                 2.31

juhlee-microsoft commented on XML/TestCases/NestedKvmTests.xml:
```diff
@@ -12,6 +12,7 @@
             <param>NestedUserPassword='NESTED_VM_PASSWORD'</param>
             <param>HostFwdPort=60022</param>
         </TestParameters>
+        <Timeout>720</Timeout>
```
This test directly calls Bash script, and I can not find timeout value in nested_kvm_basic_test.sh. Instead, there is 'retry_times' variable with value 24. Help me understand this test.

iamshital replied:
@juhlee-microsoft 
https://github.com/LIS/LISAv2/blob/master/Libraries/Test-Framework.psm1#L485

iamshital replied:
@lizzha Is this timeout safe for smaller Azure VM sizes also? (1 core / 2 core VMs)

lizzha replied:
yes, the minimum size supported is Standard_E2_v3, I've verified it's OK

juhlee-microsoft replied:
acked

adriansuhov committed:
SR-IOV: XML refresh

The test case names and order was refreshed. New TC names are:
SRIOV-VERIFY-LSPCI
SRIOV-VERIFY-SINGLE-VF-CONNECTION
SRIOV-VERIFY-SINGLE-VF-CONNECTION-MAX-VCPU
SRIOV-VERIFY-MAX-VF-CONNECTION
SRIOV-VERIFY-MAX-VF-CONNECTION-MAX-VCPU
SRIOV-ADD-MAX-NICS-DURING-PROVISION
SRIOV-ADD-MAX-NICS-AFTER-PROVISION
SRIOV-DISABLE-ENABLE-AN
SRIOV-INTERRUPTS-CHANGE
SRIOV-DISABLEVF-ON-GUEST
SRIOV-RELOAD-MODULE
SRIOV-IPERF-STRESS

During this refresh, the script "verify-ifup-nics.sh" was removed and
it's features were integrated in NET-Max-NICs.sh. By doing this, we'll
be using the same bash script for Azure and Hyper-V testing Max NICs
test cases.
